### PR TITLE
OSDOCS-9206 Enabling Azure storage account encryption

### PR DESCRIPTION
**Pull request by bscott-rh**
Version(s):
4.15

Issue:
https://issues.redhat.com/browse/OSDOCS-9206

Link to docs preview:
https://70663--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_azure/installation-config-parameters-azure#installation-configuration-parameters-additional-azure_installation-config-parameters-azure

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by bscott-rh** [comment]
@jinyunma @rna-afk Here's the PR adding the configuration parameters for Azure storage encryption. Can you check them for accuracy and see if I am missing any?

I am considering adding a section to the [existing doc](https://docs.openshift.com/container-platform/4.15/installing/installing_azure/enabling-user-managed-encryption-azure.html) to show that there are two separate encryption features available. WDYT? Is it enough to have the parameters in the table and an entry in the release notes? Thank you

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -1826,6 +1826,45 @@ For more information about the support scope of Red Hat Technology Preview featu
...
+      keyVault:
+        resourceGroup:
+|Specifies the name of the resource group that contains the key vault and managed identity.
+|String.
+
+|platform:
+  azure:
+    customerManagedKey:
+      keyVault:
+        subscriptionId:
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1857]: This parameter has been removed.
```
$ ./openshift-install explain installconfig.platform.azure.customerManagedKey.keyVault
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <object>
  KeyVault is the keyvault used for the customer created key required for encryption.

FIELDS:
    keyName <string> -required-
      KeyName is the name of the key vault key.

    name <string> -required-
      Name is the name of the key vault.

    resourceGroup <string> -required-
      ResourceGroup defines the Azure resource group used by the key vault.

```

**Comment by jinyunma** [comment]
> [existing doc](https://docs.openshift.com/container-platform/4.15/installing/installing_azure/enabling-user-managed-encryption-azure.html)

This doc mainly talks about creating disk encryption set for os disk/data disk encryption, need more procedures. While new parameters introduced in this PR just only encrypting azure storage account used by installer ( just need key in keyvault and one user-managed identity), I'm not sure if it is good idea to describe it in that doc, maybe it is enough to add those parameters in the table. @rna-afk wdyt?

**Comment by rna-afk** [comment]
I'm guessing this is just the spot for updating the list of install config parameters we did for enabling the encryption and there's other docs for specifying exactly how to do it so I think this is good to go?

**Comment by bscott-rh** [comment]
> I'm guessing this is just the spot for updating the list of install config parameters we did for enabling the encryption and there's other docs for specifying exactly how to do it so I think this is good to go?

That is the question - whether you think we should add a procedure on how to enable the encryption or if you feel having the parameters and the release note is sufficient

**Comment by rna-afk** [comment]
There's not a lot to do for encryption. If the user has the key generated in a key vault, they should be able to pass it to the installer using these fields and the encryption is done so I think this is fine.

**Pull request merged**